### PR TITLE
[C-2527] adds preference widget component

### PR DESCRIPTION
**Pull request by suhasdeshpande**
## Description
```typescript
export function ConsumerComponent(): React.ReactElement {
  const preferenceTemplateId = "YOUR_TEMPLATE_ID";
  const { CLIENT_KEY, USER_ID } = {CLIENT_KEY:"YOUR_CLIENT_KEY", USER_ID: "YOUR_RECIPIENT_ID"};
  return (
    <CourierProvider clientKey={CLIENT_KEY} userId={USER_ID}>
      <Preferences preferenceTemplateId={preferenceTemplateId}></Preferences>
    </CourierProvider>
  );
}
```

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: packages/components/src/index.html [hunk 1]
```diff
@@ -13,7 +13,6 @@
     </style>
     <script type="text/javascript">
       window.courierConfig = {
-        clientKey: "ZDRiMTViNmMtODdmZi00OWMzLWI5YTgtYmU5NTc3YTRlM2Y1",
```
[the remark this answers is not in the document]
Comment (rileylnapier) [reply packages/components/src/index.html:16]: should be env variable

[the remark this answers is not in the document]
Comment (rileylnapier) [reply packages/components/src/index.html:16]: we need a clientkey

File: packages/react-preferences/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "@trycourier/react-preferences",
+  "version": "0.0.1-dev",
```
Comment (rileylnapier) [inline packages/react-preferences/package.json:3]: should match the other packages

File: packages/react-preferences/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "main": "dist/index.js",
+  "types": "typings/index.d.ts",
+  "scripts": {
+    "babel": "babel src -d dist --extensions \".ts,.tsx\" --ignore \"src/**/__tests__/**\"",
+    "build:watch": "yarn babel --watch",
+    "build": "rimraf dist && yarn babel",
+    "clean": "rimraf dist",
+    "test": "jest -c jest.config.js --runInBand --silent",
+    "type-check": "tsc --noEmit",
```
Comment (rileylnapier) [inline packages/react-preferences/package.json:12]: missing "readme" script

File: packages/react-preferences/src/components/Status.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  handleOnPreferenceChange,
+}) => (
+  <StyledToggle>
+    <label>
+      <span>{label}</span>
+      <Toggle
+        icons={false}
+        checked={value === "OPTED_IN" ? true : false}
+        value={value}
+        onChange={() =>
```
Comment (rileylnapier) [inline packages/react-preferences/src/components/Status.tsx:18]: can we not use anon functions in the render?

File: packages/react-preferences/src/components/StyledToggle.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
+import styled from "styled-components";
+
+export const StyledToggle = styled.div`
```
Comment (rileylnapier) [inline packages/react-preferences/src/components/StyledToggle.tsx:3]: do you think we will want to reuse this in other components?  i wonder if we need a common package?  prob ok for now

File: packages/react-preferences/src/components/preferences.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+}> = ({ preferenceTemplateId }) => {
+  const [preferenceGroup, updatePreferenceGroup] = usePreferenceTemplate(
+    preferenceTemplateId
+  );
+
+  if (!preferenceGroup) {
+    return null;
+  }
+
+  const handleOnPreferenceChange = (...args) => {
```
Comment (rileylnapier) [inline packages/react-preferences/src/components/preferences.tsx:31]: is there a reason to spread the args?

File: packages/react-preferences/package.json (hunk 2 above)
Comment (suhasdeshpande) [reply packages/react-preferences/package.json:3]: We want to match the exact version? or you are suggesting to use `-alpha` suffix?

File: packages/react-preferences/src/components/StyledToggle.tsx (hunk 5 above)
Comment (suhasdeshpande) [reply packages/react-preferences/src/components/StyledToggle.tsx:3]: probably not right away, but will move it to common package if that changes

File: packages/components/src/index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (suhasdeshpande) [reply packages/components/src/index.html]: added that back. I can investigate env variables

File: packages/react-inbox/README.md [hunk 7]
```diff
@@ -1,129 +0,0 @@
-<!-- START doctoc generated TOC please keep comment here to allow auto update -->
-<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
-
-- [What is Inbox?](#what-is-inbox)
```
Comment (suhasdeshpande) [inline packages/react-inbox/README.md:4]: I don't know how this got deleted.

File: packages/react-preferences/package.json (hunk 3 above)
Comment (suhasdeshpande) [reply packages/react-preferences/package.json:12]: could we add readme when after we are #69?

**Review by rileylnapier — DISMISSED** [review]

File: packages/react-preferences/package.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "@trycourier/react-preferences",
+  "version": "0.1.10-alpha.27",
```
Comment (suhasdeshpande) [inline packages/react-preferences/package.json:3]: updated the version to match with rest of the packages @rileylnapier

**Review by rileylnapier — APPROVED** [review]

**Pull request merged**